imap/mailbox.c

```diff
@@ -1829,12 +1829,46 @@ EXPORTED struct entryattlist *mailbox_extract_annots(const struct mailbox *mailb
     return annots;
 }
 
+struct index_header_offsets {
+    off_t mailbox_options;
+    off_t leaked_cache;
+    off_t highestmodseq;
+    off_t deletedmodseq;
+    off_t exists;
+    off_t changes_epoch;
+    off_t quota_deleted_used;
+    off_t quota_expunged_used;
+};
+
+static const struct index_header_offsets orig_offsets = {
+    ORIG_OFFSET_MAILBOX_OPTIONS,
+    ORIG_OFFSET_LEAKED_CACHE,
+    ORIG_OFFSET_HIGHESTMODSEQ,
+    ORIG_OFFSET_DELETEDMODSEQ,
+    ORIG_OFFSET_EXISTS,
+    ORIG_OFFSET_CHANGES_EPOCH,
+    ORIG_OFFSET_QUOTA_DELETED_USED,
+    ORIG_OFFSET_QUOTA_EXPUNGED_USED
+};
+
+static const struct index_header_offsets new_offsets = {
+    OFFSET_MAILBOX_OPTIONS,
+    OFFSET_LEAKED_CACHE,
+    OFFSET_HIGHESTMODSEQ,
+    OFFSET_DELETEDMODSEQ,
+    OFFSET_EXISTS,
+    OFFSET_CHANGES_EPOCH,
+    OFFSET_QUOTA_DELETED_USED,
+    OFFSET_QUOTA_EXPUNGED_USED
+};
```

Comment:
Hmm... I wonder what this will look like in the future when we rearrange the offsets again, and now need to keep track of n+1 sets

Reply:
Yeah, good thought.